libraries/aws_config_recorder.rb

```diff
@@ -13,46 +13,51 @@ class AwsConfigurationRecorder < AwsResourceBase
       it { should have_include_global_resource_types }
     end
   "
-  attr_reader :role_arn, :resource_types, :recorder_name, :exists, :recording_all_resource_types, :recording_all_global_types
-  alias exists? exists
+  attr_reader :role_arn, :resource_types, :recorder_name, :recording_all_resource_types, :recording_all_global_types
   alias recording_all_resource_types? recording_all_resource_types
   alias recording_all_global_types? recording_all_global_types
 
   def initialize(opts = {})
-    opts = { recorder_name: opts } if opts.is_a?(String) # this preserves the original scalar interface
+    opts = { recorder_name: opts } if opts.is_a?(String)
+
     super(opts)
-    validate_parameters([:recorder_name]) if !opts.nil?
-    @recorder_name = opts[:recorder_name] unless opts.nil?
-    query = @recorder_name ? { configuration_recorder_names: [@recorder_name] } : {}
+    validate_parameters(allow: [:recorder_name])
+
+    query = !opts.nil? && opts.key?(:recorder_name) ? { configuration_recorder_names: [opts[:recorder_name]] } : {}
 
     catch_aws_errors do
       begin
-        @resp = @aws.config_client.describe_configuration_recorders(query)
-        raise ArgumentError, 'Error: unexpectedly received multiple AWS Config Recorder objects from API; expected to be singleton per-region. ' if @resp.configuration_recorders.count > 1
-        @exists = !@resp.configuration_recorders.empty?
+        resp = @aws.config_client.describe_configuration_recorders(query)
+        raise ArgumentError, "#{@__resource_name__}: unexpectedly received multiple AWS Config Recorder objects from API; expected to be singleton per-region." if resp.configuration_recorders.count > 1
       rescue Aws::ConfigService::Errors::NoSuchConfigurationRecorderException
-        @exists = false
```

Comment:
Worth checking the `exists` issue as mentioned above.

Reply:
Ci Passing on this